ta_services/secure_key_services/src/handle.c

```diff
@@ -62,22 +67,43 @@ uint32_t handle_get(struct handle_db *db, void *ptr)
 	return n;
 }
 
+/*
+ * Protect from speculative access based on out bound handle values:
+ * Generate a validation mask from signed difference from max valid handle
+ * value to argument handle value:
+ * - out bound handle values generate a null mask.
+ * - handle values that can be seen as negative index generate a null mask.
+ * This scheme expects value 0 is an invalid handle value and should
+ * always return an invalid reference: here always a NULL reference.
```

Comment:
This description is not that clear. 
"here always a NULL reference" refers how handle is used, not this specific function.

Reply:
I'd rather implement `array_index_mask_nospec()` i.e., the same interface as the kernel's, because there is no reason to do differently.

Reply:
i would rather implement it our own way than reusing linux kernel code snipet in optee_os.